pkg/kubectl/cmd/drain.go

```diff
@@ -466,7 +466,7 @@ func (o *DrainOptions) getPodsForDeletion(nodeInfo *resource.Info) (pods []corev
 
 	for _, pod := range podList.Items {
 		podOk := true
-		for _, filt := range []podFilter{mirrorPodFilter, o.localStorageFilter, o.unreplicatedFilter, o.daemonsetFilter} {
+		for _, filt := range []podFilter{o.daemonsetFilter, mirrorPodFilter, o.localStorageFilter, o.unreplicatedFilter} {
```

Comment:
Ok, if I'm understanding this filter chain, the idea is that you'll fail the command on a pod that you're trying to evict that can't safely be evicted.  Because daemonsets live and die with nodes, you don't really care if the pod doesn't fail over.

Rather trying to bake knowledge of various workload controllers into this command (which won't as third parties start creating these concepts), why not create an annotation or label that indicates that it is ok to kill a particular pod?  DaemonSets and any other workload controller in that category could describe its pods that way.  Since it is strictly downgrading reliability, there's no need to secure such an annotation or label.

Reply:
Proposal for this: https://docs.google.com/document/d/1t7nuQVouwEnYgNaZBbfk_HJXqAGEPQ6EC0d7Ry1ml3Q/edit#heading=h.ke1hbaqugq1u